ws.go

```
@@ -0,0 +1,130 @@
+		// gorilla/websocket.Upgrader.Upgrade 会自动处理。
+		// 它会检查请求是 HTTP/1.1 Upgrade 还是 HTTP/2 CONNECT with :protocol=websocket。
+
+		// 对于 HTTP/2，Upgrade() 方法不会发送 101，而是处理 CONNECT 的 200 OK。
+		// 它也不会调用 Hijack，因为连接已经在 HTTP/2 流上。
+		conn, err := upgrader.Upgrade(c.Writer, c.Request, nil)
+		if err != nil {
+			// 升级失败。gorilla/websocket.Upgrade 会处理错误响应的发送。
+			// (对于 HTTP/1.1 会是 400/403 等；对于 HTTP/2 也是类似的非 2xx 响应)
+			var httpErr websocket.HandshakeError
+			statusCode := http.StatusBadRequest // 默认
+			if errors.As(err, &httpErr) {
```

golangci-lint (1.64.8)

SA9003: empty branch

(staticcheck)